activerecord/lib/active_record/attributes.rb

```diff
@@ -264,6 +264,7 @@ def load_schema! # :nodoc:
         super
         attributes_to_define_after_schema_loads.each do |name, (cast_type, default)|
           cast_type = cast_type[type_for_attribute(name)] if Proc === cast_type
+          raise UnknownAttributeError.new(self, name) if Enum::EnumType === cast_type && !columns_hash.key?(name)
```

Comment:
Ideally we'd do this from the `Enum` module. You can redefine `load_schema!` there and call `super`.